Certbot authenticator (#2328)
kyma-project/test-infra: development/tools/cmd/certbotauthenticator/README.md

#certbotauthenticator

##Overview

Certbotauthenticator is a binary called by certbot during certificate generation process. It's used in phase of manual DNS challenge authentication. Cerbot in manual mode is passing domain name and authentication token as environment variables to the certbotauthenticator, for the purpose os creating TXT record in a domain, to let LetsEncrypt system validate domain ownership. After validation, certbotauthenticator is called again to clean TXT records created for validation.

> ##Justification
> 
> We made this tool, becasue certbot dns-google plugin doesn't support creating TXT records within DNS Zone existing in different project, than service account used to authenticate in GCP.

Reviewer: ```suggestion
## Justification

We have created this tool because the certbot dns-google plugin does not support creating TXT records within a DNS zone existing in different project than the service account used to authenticate in GCP.
```

## Usage

### Environment variables

> In manual mode, certbot passes these data as environment variables when calling external tools for authentication and clean-up.

Reviewer: ```suggestion
In manual mode, certbot passes this data as environment variables when calling external tools for authentication and clean-up.
```

## Justification

We have created this tool because the certbot dns-google plugin does not support creating TXT records within a DNS zone existing in different project than the service account used to authenticate in GCP.

## Usage

### Environment variables

In manual mode, certbot passes these data as environment variables when calling external tools for authentication and clean-up.

- **CERTBOT_VALIDATION** - validation token, expected by Let's Enrypt as a record value.
- **CERTBOT_DOMAIN** - domain name against which Let's Encrypt will execute validation.
- **CERTBOT_AUTH_OUTPUT** - the output from the authentication step, passed only to call during the clean-up.

### Authentication
Use the **GOOGLE_APPLICATION_CREDENTIALS** environment variable to provide credentials for authentication in GCP.

### CLI parameters

The certbotauthenticator accepts the following command line parameters:

> |Parameter | Description | Value|

Reviewer: ```suggestion
|Parameter | Description | Default value|
```
